docs/guide/docs/infrastructure/development-pipeline.md

```diff
@@ -0,0 +1,114 @@
+
+### Setting up Botpress Git Syncing (with GitHub)
+
+1. [Download Botpress](https://botpress.com/download)
+2. Open your terminal and ```cd [PATH]/botpress-v12_20_2-darwin-x64```
+3. Set up GitHub remote tracking
+```
+git init
+git add README.md
+git commit -m "first commit"
+git branch -M main
+git remote add origin git@github.com:[GH_USERNAME]/bp-project.git
+git push -u origin main
+```
+4. You can add a develop and staging branch to fit your workflow
+5. Everytime you make a change in the Conversation Studio or in your preferred code editor, you can commit that change to track changes.
+6. When you are ready to send you chatbot to production, you can ```git merge``` your develop branch in master.
+7. Then, you ```checkout master``` and you export your chatbot to upload it on your production environment (usually cloud deployment)
+
+### Syncing Changes Between Environments
+
+We will use Git to sync changes between 2 environments and promote an environment (i.e., promote staging to production).
+
+Let's assume that given a pipeline with three environments, **development**, **staging**, and **production**, there are some changes both on production and staging, and you want to promote staging to production. What we want to do is the following:
+
+1. Create a merge conflict so we can choose what we want in a merge conflict tool.
+2. Resolve conflicts (i.e., merge staging into production)
+3. Push the results to master so they are deployed to your production environment.
+
+First, create a branch and sync it with the production environment:
+
+```
+git checkout master && git checkout -b prod-sync
+./bp pull --url {PROD_SERVER_URL} --authToken {YOUR_AUTH_TOKEN} --targetDir {TARGET_DIRECTORY}
+git commit -am 'sync prod'
+```
+
+Repeat the process with staging environment:
+
+```
+git checkout master && git checkout -b staging-sync
+./bp pull --url {STAGING_SERVER_URL} --authToken {YOUR_AUTH_TOKEN} --targetDir {TARGET_DIRECTORY}
+git commit -am 'sync staging'
+```
+
+Then merge the staging changes into the prod changes:
+
+`git checkout prod-sync && git merge staging-sync`
+
+This will create a merge conflict; use your preferred merge tool to review the changes and resolve the conflicts. Once done, you can publish your branch and create a pull request (if your hosted git allows it), and merge it to master.
+
+Once your master branch is up-to-date, you'll be able to [push](versions#push) the changes to production with:
+
+`./bp push --url {PROD_SERVER_URL} --authToken {YOUR_AUTH_TOKEN} --targetDir {TARGET_DIRECTORY}`
+
+With these quick tips, you can now promote any environment changes to any stage in your deployment pipeline.
+
+## Editing on the Production Server
+
+Once your bot is deployed, you (and non-technical team members) **can still make changes to your bots from Botpress Studio**, which is one significant advantage of using Botpress. This is made possible by our built-in versioning system.
```

Comment:
I'm not sure we want to promote editing chatbots in production anymore as this differs very much of our vision and from a normal developer workflow. 

CC @spg I'd love your thoughts on this whole doc section

Reply:
Yes, doing changes in production is discouraged

Reply:
@spg @EFF What are you guys recommending? We add a mention ''Not recommended'' or we remove all mention of that altogether.

Reply:
@marcmerce  @Michael-N-M  I would remove this section altogether.